code/processes/rdb.q

```diff
@@ -199,6 +199,9 @@ reload:.rdb.reload
 /-set the partition that is held in the rdb (for use by the gateway)
 .rdb.setpartition[]
 
+/-rounds GMT offset to nearest hour
+localoffset: {01:00*"j"$(`minute$x)%60}(.proc.cp[]-.z.p)
+
 /-change timeout to zero before eod flush
-.timer.repeat[.eodtime.nextroll-00:01;0W;1D;
+.timer.repeat[.eodtime.nextroll+localoffset-00:01;0W;1D;
```

Comment:
Can you do this without creating the localoffset variable? I assume it isn't used anywhere else. You also need to account for time differences which aren't a full hour (e.g. indian standard time)